core/cas-server-core-logout-api/src/main/java/org/apereo/cas/logout/slo/DefaultSingleLogoutServiceMessageHandler.java

```diff
@@ -23,4 +27,21 @@ public DefaultSingleLogoutServiceMessageHandler(final HttpClient httpClient,
         super(httpClient, logoutMessageBuilder, servicesManager, singleLogoutServiceLogoutUrlBuilder,
             asynchronous, authenticationRequestServiceSelectionStrategies);
     }
+
+    /**
+     * The default handler only applies on the CAS protocol services.
+     * For the SAML and OIDC protocols, specific handlers exist.
+     * For the OAuth protocol, no logout process is handled.
+     *
+     * The check of the service type is made by the class name not to add support-* dependencies.
+     *
+     * @param singleLogoutService the single logout service
+     * @param registeredService   the registered service
+     * @return whether the handler applies
+     */
+    @Override
+    protected boolean supportsInternal(final WebApplicationService singleLogoutService, final RegisteredService registeredService) {
+        val name = registeredService.getClass().getSimpleName();
+        return !"SamlRegisteredService".equals(name) && !"OidcRegisteredService".equals(name) && !"OAuthRegisteredService".equals(name);
```

Comment:
This should be unnecessary. `SingleLogoutServiceMessageHandler` are or can be ordered. The default runs last. So if all other handlers fail to support the service, this one should run by default automatically, and there is need to check for service types as such, because other handlers should be given more priority with a lower order value.

Also, comparing types by String will never scale and unmaintainable. (This is already missing a service type, for example)

Reply:
In general the order of execution is:

1. Protocol-specific ones, such as SAML2, OIDC, OAUTH, etc
1.1 Each handler checks for a service type and passes it onto the next, if it doesn't support the type
2. Default runs as a catch-all.

Reply:
Yes, it makes sense. It will be much better.